### ISSUE TITLE
Wrong HOST when calling `blob.get_signed_url` in local `api_root`

### ISSUE THREAD

**Issue by erincar** [talkiq/gcloud-aio#828]
https://github.com/talkiq/gcloud-aio/blob/e738eb839239333a9fe69a448b51ef0fc0e4a4a1/storage/gcloud/aio/storage/blob.py#L34

Hello,

I am running `fake-gcs-server` in my local compose stack and connect to it via `gcloud.aio.storage.Storage`.
I set up the `api_root` via `STORAGE_EMULATOR_HOST`, and can see that the correct one is used.

However, when generating a signed URL for a blob, I receive one starting with `storage.googleapis.com`.
I am suspecting that is due to the hardcoded `HOST` referred above.

Is this usage intended? Or should the `HOST` be pointing to the `api_root` instead?

**Comment by TheKevJames** [comment]
Yes, good catch, this was an oversight! Will include this fix in the next `storage` release.
